server/src/main/java/ru/spbau/blackout/server/RoomServer.java

```diff
@@ -3,51 +3,67 @@
 import java.net.ServerSocket;
 import java.net.Socket;
+import java.util.ArrayList;
 import java.util.Deque;
+import java.util.List;
 import java.util.concurrent.ConcurrentLinkedDeque;
 import java.util.concurrent.atomic.AtomicInteger;
 
-public class RoomServer {
+class RoomServer {
+
+    private static final int PLAYERS_NUMBER = 2;
 
     private final int port;
-    private final Deque<RoomClientThread> roomClientThreads = new ConcurrentLinkedDeque<>();
+    private final Deque<ClientThread> clientThreads = new ConcurrentLinkedDeque<>();
     private final AtomicInteger playersNumber = new AtomicInteger();
 
-    public AtomicInteger getPlayersNumber() {
-        return playersNumber;
+     RoomServer(int port) {
+        this.port = port;
     }
 
-    public RoomServer(int port) {
-        this.port = port;
+    int getPlayersNumber() {
+        return playersNumber.get();
     }
 
-    public void run() {
+    @SuppressWarnings("InfiniteLoopStatement")
+    void run() {
         try (ServerSocket serverSocket = new ServerSocket(port)) {
-            while (true) {
+            do {
                 final Socket nextSocket = serverSocket.accept();
-                final RoomClientThread nextThread = new RoomClientThread(this, nextSocket);
-                synchronized (System.out) {
-                    System.out.println("New thread for a connection is created");
-                }
-                roomClientThreads.add(nextThread);
+                final ClientThread nextThread = new ClientThread(this, nextSocket);
+                log("New thread for a connection is created.");
+                clientThreads.add(nextThread);
                 playersNumber.addAndGet(1);
                 nextThread.start();
-            }
+                maybePlayGame(PLAYERS_NUMBER);
+            } while (true);
         } catch (IOException e) {
-            synchronized (System.out) {
-                System.out.println("Exception caught when trying to listen on port "
-                        + port + " or listening for a connection");
-                System.out.println(e.getMessage());
+            log("Exception caught when trying to listen on port " + port +
+                " or listening for a connection:" + e.getMessage());
+        }
+    }
+
+    private synchronized void maybePlayGame(int playersForGame) {
+        if (playersNumber.get() >= playersForGame) {
+            playersNumber.addAndGet(-playersForGame);
+            final List<ClientThread> clients = new ArrayList<>(playersForGame);
+            for (int playerIndex = 0; playerIndex < playersForGame; playerIndex++) {
+                clients.add(clientThreads.removeFirst());
             }
+            final Thread game = new Game(this, clients);
+            game.start();
         }
     }
 
-    void discard(RoomClientThread clientThread) {
-        roomClientThreads.remove(clientThread);
-        playersNumber.addAndGet(-1);
+    void discard(ClientThread clientThread) {
```

Comment:
It seems reasonable, thank you.
But how will it help me to avoid any synchronization tools if several threads at once may want to do cleanup(if I'll choose doing cleanup right here)?
finally { <do cleanup  //, or, better, notify the main thread that cleanup is necessary// > }
}